crates/sui-replay/src/data_fetcher.rs

```diff
@@ -806,4 +815,12 @@ impl DataFetcher for NodeStateDumpFetcher {
     async fn get_chain_id(&self) -> Result<String, ReplayEngineError> {
         unimplemented!("get_chain_id for state dump is not implemented")
     }
+
+    async fn get_child_object(
+        &self,
+        _object_id: &ObjectID,
+        _version_upper_bound: VersionNumber,
+    ) -> Result<Object, ReplayEngineError> {
+        unimplemented!("get child object is not implemented for state dump");
+    }
```

Comment:
are we sure that in this case the child objects are properly loaded from the state dump and this fn isn't called in that case?

Reply:
yes, for a dump, child objects will be loaded the same way they used to and this function won't be called. The result will be served from prepopulated local storage directly